src/js/types/translation.ts

```diff
@@ -1,2 +1,2 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-export type NamespaceTranslationFunction = (key: string, options?: any) => string;
+export type NamespaceTranslationFunction = (key: string | undefined, options?: any) => string;
```

Comment:
why this type change?

Reply:
It can handle undefined. Reduces duplicated error handling logic in undefined / string situations.